var/spack/repos/builtin/packages/quantum-espresso/package.py

```diff
@@ -111,6 +112,8 @@ class QuantumEspresso(CMakePackage):
     with when("+mpi"):
         depends_on("mpi")
         variant("scalapack", default=True, description="Enables scalapack support")
+        # provide Scalapack, Cluster FFT and Cluster Sparse Solver
+        depends_on("intel-oneapi-mkl+cluster", when="^intel-oneapi-mkl")
```

Comment:
QE doesn't use cluster FFT and sparse solver. If  I remember correctly, scalapack is available without `+cluster`

Reply:
I understood Scalapack are included in the cluster edition from [here](https://www.intel.com/content/www/us/en/develop/documentation/onemkl-macos-developer-guide/top/working-with-onemkl-cluster-software.html) and from the recipe  

```
sed -n 90p spack/var/spack/repos/builtin/packages/intel-oneapi-mkl/package.py
provides("scalapack", when="+cluster")
```

Reply:
In my `intel-oneapi-mkl~cluster` installation, I saw scalapack available.
```
opt/spack/linux-ubuntu22.04-skylake_avx512/gcc-9.4.0/intel-oneapi-mkl-2022.2.0-naqimuwsngvepi3qkm2zcmrjiapvpznz/mkl/latest/lib/intel64$ ls -lrt *scalapack*
-rw-r--r-- 1 yeluo QMCPACK_dev 12334132 Aug  5 02:49 libmkl_scalapack_lp64.a
-rwxr-xr-x 1 yeluo QMCPACK_dev  7718648 Aug  5 02:49 libmkl_scalapack_ilp64.so.2
-rw-r--r-- 1 yeluo QMCPACK_dev 12244638 Aug  5 02:49 libmkl_scalapack_ilp64.a
-rwxr-xr-x 1 yeluo QMCPACK_dev  7736496 Aug  5 02:49 libmkl_scalapack_lp64.so.2
lrwxrwxrwx 1 yeluo QMCPACK_dev       26 Sep 16 16:50 libmkl_scalapack_lp64.so -> libmkl_scalapack_lp64.so.2
lrwxrwxrwx 1 yeluo QMCPACK_dev       27 Sep 16 16:50 libmkl_scalapack_ilp64.so -> libmkl_scalapack_ilp64.so.2
```
I feel better to have mkl always providing scalapack.

Reply:
I confirm after `spack install quantum-espresso ^intel-oneapi-mkl` that actually scalapack are found and used to build quantum espresso:
```
-- Found SCALAPACK: /g100/home/userinternal/lbellen1/spack/opt/spack/linux-centos8-skylake_avx512/gcc-8.4.1/intel-oneapi-mkl-2022.1.0-3sefpr33gex5a3aujt6caz6cfqpb53zw/mkl/2022.1.0/lib/intel64/libmkl_scalapack_lp64.so;/g100/home/userinternal/lbellen1/spack/opt/spack/linux-centos8-skylake_avx512/gcc-8.4.1/intel-oneapi-mkl-2022.1.0-3sefpr33gex5a3aujt6caz6cfqpb53zw/mkl/2022.1.0/lib/intel64/libmkl_blacs_openmpi_lp64.so;/g100/home/userinternal/lbellen1/spack/opt/spack/linux-centos8-skylake_avx512/gcc-8.4.1/intel-oneapi-mkl-2022.1.0-3sefpr33gex5a3aujt6caz6cfqpb53zw/mkl/2022.1.0/lib/intel64/libmkl_gf_lp64.so;/g100/home/userinternal/lbellen1/spack/opt/spack/linux-centos8-skylake_avx512/gcc-8.4.1/intel-oneapi-mkl-2022.1.0-3sefpr33gex5a3aujt6caz6cfqpb53zw/mkl/2022.1.0/lib/intel64/libmkl_gnu_thread.so;/g100/home/userinternal/lbellen1/spack/opt/spack/linux-centos8-skylake_avx512/gcc-8.4.1/intel-oneapi-mkl-2022.1.0-3sefpr33gex5a3aujt6caz6cfqpb53zw/mkl/2022.1.0/lib/intel64/libmkl_core.so;/usr/lib/gcc/x86_64-redhat-linux/8/libgomp.so;-lm;-ldl;
```
However, `quantum-espresso ^intel-oneapi-mkl` is concretized also with `netlib-scalapack`, which is installed but not used.  Shouldn't we inhibit this behavior with the +cluster variant?

Reply:
I made the provides conditional because I was concerned about the case where you want another provider, but since intel-mkl is unconditional, I will change it.

Reply:
#33378 changes intel-oneapi-mkl to provide scalapack even if you don't do +cluster